### PR TITLE
Adjust headers with deheader

### PR DESCRIPTION
**Pull request by remuswu1019**
Add required headers and remove unused headers

Fixes:#1444

### CODE REVIEW

File: lib/tpm2.c [hunk 1]
```diff
@@ -1,13 +1,10 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
 
-#include <tss2/tss2_esys.h>
 #include <tss2/tss2_mu.h>
 
 #include "log.h"
 #include "object.h"
-#include "tpm2.h"
```
Comment (AndreasFuchsSIT) [inline lib/tpm2.c]: keeping the <thisfile>.h on includes is a good idea ensure alignment of function declarations in header and implementations.

File: lib/tpm2_attr_util.c [hunk 2]
```diff
@@ -1,14 +1,11 @@
...
 
 #include <stdbool.h>
+#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
-#include <tss2/tss2_sys.h>
-
 #include "log.h"
-#include "tpm2_attr_util.h"
```
Comment (AndreasFuchsSIT) [inline lib/tpm2_attr_util.c]: again, corresponding header...

File: lib/tpm2_auth_util.c [hunk 3]
```diff
@@ -2,23 +2,17 @@
...
 #include <unistd.h>
 
-#include <tss2/tss2_esys.h>
-#include <tss2/tss2_mu.h>
-
 #include "files.h"
 #include "log.h"
 #include "pcr.h"
 #include "tpm2.h"
-#include "tpm2_auth_util.h"
```
Comment (AndreasFuchsSIT) [inline lib/tpm2_auth_util.c]: again

File: lib/tpm2_capability.c [hunk 4]
```diff
@@ -4,13 +4,8 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <tss2/tss2_esys.h>
-
 #include "log.h"
 #include "tpm2.h"
-#include "tpm2_util.h"
-
-#include "tpm2_capability.h"
```
Comment (AndreasFuchsSIT) [inline lib/tpm2_capability.c]: again

File: lib/tpm2.c (hunk 1 above)
Comment (remuswu1019) [reply lib/tpm2.c]: I am not sure why deheader remove this header.
I will add them back and commit again, Thanks.

File: lib/tpm2_cc_util.c [hunk 5]
```diff
@@ -1,8 +1,8 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
+
 #include <string.h>
 
 #include "log.h"
-#include "tpm2_cc_util.h"
```
Comment (remuswu1019) [inline lib/tpm2_cc_util.c]: keeping the .h

File: lib/tpm2_ctx_mgmt.c [hunk 6]
```diff
@@ -1,14 +1,6 @@
-#include <stdbool.h>
-
-#include <tss2/tss2_esys.h>
-
 #include "log.h"
-#include "object.h"
 #include "tpm2.h"
 #include "tpm2_auth_util.h"
-#include "tpm2_ctx_mgmt.h"
```
Comment (remuswu1019) [inline lib/tpm2_ctx_mgmt.c]: keeping the .h

File: lib/tpm2_error.c [hunk 7]
```diff
@@ -1,13 +1,9 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
 
 #include <stdarg.h>
-#include <stdbool.h>
 #include <stdio.h>
 #include <string.h>
 
-#include <tss2/tss2_sys.h>
-
-#include "tpm2_error.h"
```
Comment (remuswu1019) [inline lib/tpm2_error.c]: keeping the .h

File: lib/tpm2_hash.c [hunk 8]
```diff
@@ -1,15 +1,12 @@
...
+#include <stdio.h>
 #include <string.h>
 
-#include <tss2/tss2_esys.h>
-
-#include "log.h"
 #include "files.h"
+#include "log.h"
 #include "tpm2.h"
-#include "tpm2_hash.h"
```
Comment (remuswu1019) [inline lib/tpm2_hash.c]: keeping the .h

File: lib/tpm2_hierarchy.c [hunk 9]
```diff
@@ -4,13 +4,9 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <tss2/tss2_esys.h>
-
 #include "log.h"
 #include "tpm2.h"
 #include "tpm2_auth_util.h"
-#include "tpm2_hierarchy.h"
```
Comment (remuswu1019) [inline lib/tpm2_hierarchy.c]: keeping the .h

File: lib/tpm2_identity_util.c [hunk 10]
```diff
@@ -1,30 +1,15 @@
...
-
 #include <tss2/tss2_mu.h>
-#include <tss2/tss2_sys.h>
 
-#include "files.h"
 #include "log.h"
 #include "tpm2_alg_util.h"
 #include "tpm2_kdfa.h"
 #include "tpm2_openssl.h"
-#include "tpm2_identity_util.h"
```
Comment (remuswu1019) [inline lib/tpm2_identity_util.c]: keeping the .h

File: lib/tpm2_kdfa.c [hunk 11]
```diff
@@ -1,14 +1,7 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
 
-#include <tss2/tss2_sys.h>
-
- #include <openssl/err.h>
-#include <openssl/hmac.h>
-
 #include "log.h"
-#include "tpm2_kdfa.h"
```
Comment (remuswu1019) [inline lib/tpm2_kdfa.c]: keeping the .h

File: lib/tpm2_policy.c [hunk 12]
```diff
@@ -1,23 +1,14 @@
...
-
 #include "files.h"
 #include "log.h"
-#include "object.h"
 #include "tpm2.h"
-#include "tpm2_hash.h"
 #include "tpm2_alg_util.h"
-#include "tpm2_auth_util.h"
 #include "tpm2_openssl.h"
-#include "tpm2_policy.h"
```
Comment (remuswu1019) [inline lib/tpm2_policy.c]: keeping the .h

File: lib/tpm2_session.c [hunk 13]
```diff
@@ -6,15 +6,9 @@
...
 #include <string.h>
 
-#include <tss2/tss2_esys.h>
-
 #include "files.h"
 #include "log.h"
 #include "tpm2.h"
-#include "tpm2_kdfa.h"
-#include "tpm2_alg_util.h"
-#include "tpm2_session.h"
```
Comment (remuswu1019) [inline lib/tpm2_session.c]: keeping the .h

File: lib/tpm2_tcti_ldr.c [hunk 14]
```diff
@@ -1,15 +1,11 @@
...
-#include <stdlib.h>
 #include <stdio.h>
-#include <dlfcn.h>
-
-#include <tss2/tss2_sys.h>
+#include <stdlib.h>
 
-#include "config.h"
 #include "log.h"
-#include "tpm2_tcti_ldr.h"
```
Comment (remuswu1019) [inline lib/tpm2_tcti_ldr.c]: keeping the .h

File: lib/tpm2_util.c [hunk 15]
```diff
@@ -1,21 +1,18 @@
...
-
-#include "log.h"
 #include "files.h"
+#include "log.h"
 #include "tpm2.h"
 #include "tpm2_alg_util.h"
 #include "tpm2_attr_util.h"
 #include "tpm2_openssl.h"
 #include "tpm2_tool.h"
-#include "tpm2_util.h"
```
Comment (remuswu1019) [inline lib/tpm2_util.c]: keeping the .h

File: test/unit/test_tpm2_session.c [hunk 16]
```diff
@@ -1,25 +1,17 @@
...
 #include <unistd.h>
 
+#include <setjmp.h>
 #include <cmocka.h>
 
-#include <tss2/tss2_esys.h>
 #include <tss2/tss2_mu.h>
 
 #include "test_session_common.h"
-#include "tpm2_session.h"
```
Comment (remuswu1019) [inline test/unit/test_tpm2_session.c]: keeping the .h

**Comment by williamcroberts** [comment]
Anything that diffstats +93 -803 is the best type of patch.

**Pull request merged**